modules/programs/aerospace.nix

```diff
@@ -48,27 +48,87 @@ in
       type = types.submodule {
         freeformType = tomlFormat.type;
         options = {
-          start-at-login = lib.mkOption {
-            type = types.bool;
+          start-at-login = mkOption {
+            type =
+              with types;
+              oneOf [
+                bool
+                (submodule {
+                  options = {
+                    enable = lib.mkEnableOption "the AeroSpace launchd service";
+                    keepAlive = mkOption {
+                      type = bool;
+                      default = true;
+                      description = ''
+                        If set to true, launchd will automatically restart AeroSpace
+                        if it crashes or is quit for any reason.
+                      '';
+                    };
+                  };
+                })
+              ];
```

Comment:
`settings` options should be generator pass through options. They should not be used for other configurations outside of the configuration file generated.